dirty_cat/_fuzzy_join.py

```diff
@@ -0,0 +1,231 @@
+"""
+Fuzzy joining tables using string columns.
+The principle is as follows:
+  1. We transform the key columns strings to vectors using CountVectorizer.
+  2. For each category, we use the nearest neighbor method to find its closest
+  neighbor and establish a match.
+  3. We match the tables using the previous information.
+Categories from the two tables that share many n-grams have greater
+probability of beeing matched together. The join is based on
+morphological similarities between strings.
+"""
+
+import pandas as pd
+import numpy as np
+from sklearn.feature_extraction.text import CountVectorizer, TfidfTransformer
+from sklearn.neighbors import NearestNeighbors
+from typing import List, Literal, Tuple
+
+
+def fuzzy_join(left_table: pd.DataFrame,
+               right_table: pd.DataFrame,
+               on: List[str],
+               return_distance: bool = False,
+               analyzer: Literal["word", "char", "char_wb"] = "char_wb",
+               ngram_range: Tuple[int, int] = (2, 4),
+               precision: Literal["nearest", "radius"] = 'nearest',
```

Comment:
I do not think that we should call this argument "precision": "precision" corresponds to something precise in stats and ML.

We could say "match_type" (but I'm not perfectly happy with it)